AppDevPods/AppDevCameraKit/ADKCamera.m

```diff
@@ -0,0 +1,1342 @@
+            focusMode = AVCaptureFocusModeLocked;
+            break;
+        case ADKCameraFocusModeAutoFocus:
+            focusMode = AVCaptureFocusModeAutoFocus;
+            break;
+        case ADKCameraFocusModeContinuousAutoFocus:
+            focusMode = AVCaptureFocusModeContinuousAutoFocus;
+            break;
+    }
+
+    NSError *error;
+    if ([self.videoCaptureDevice lockForConfiguration:&error]) {
+        if ([self.videoCaptureDevice isFocusModeSupported:focusMode]) {
+            self.videoCaptureDevice.focusMode = focusMode;
+            _cameraFocusMode = cameraFocusMode;
+        }
+        [self.videoCaptureDevice unlockForConfiguration];
+    }
+
+    if (error) {
+        [self handleErrorWithCode:error.code userInfo:error.userInfo];
+    }
+}
+
+- (void)setCameraExposureMode:(ADKCameraExposureMode)cameraExposureMode
+{
+    AVCaptureExposureMode exposureMode;
+    switch (cameraExposureMode) {
+        case ADKCameraExposureModeLocked:
+            exposureMode = AVCaptureExposureModeLocked;
+            break;
+        case ADKCameraExposureModeAutoExposure:
+            exposureMode = AVCaptureExposureModeAutoExpose;
+            break;
+        case ADKCameraExposureModeContinuousAutoExposure:
+            exposureMode = AVCaptureExposureModeContinuousAutoExposure;
+            break;
+    }
+
+    NSError *error;
+    if ([self.videoCaptureDevice lockForConfiguration:&error]) {
+        if ([self.videoCaptureDevice isExposureModeSupported:exposureMode]) {
+            self.videoCaptureDevice.exposureMode = exposureMode;
+            _cameraExposureMode = cameraExposureMode;
+        }
+        [self.videoCaptureDevice unlockForConfiguration];
+    }
+
+    if (error) {
+        [self handleErrorWithCode:error.code userInfo:error.userInfo];
+    }
+}
+
+- (void)setCameraWhiteBlanceMode:(ADKCameraWhiteBlanceMode)cameraWhiteBlanceMode
+{
+    AVCaptureWhiteBalanceMode whiteBlanceMode;
+    switch (cameraWhiteBlanceMode) {
+        case ADKCameraWhiteBlanceModeLocked:
+            whiteBlanceMode = AVCaptureWhiteBalanceModeLocked;
+            break;
+        case ADKCameraWhiteBlanceModeAutoWhiteBalance:
+            whiteBlanceMode = AVCaptureWhiteBalanceModeAutoWhiteBalance;
+            break;
+        case ADKCameraWhiteBlanceModeContinuousAutoWhiteBalance:
+            whiteBlanceMode = AVCaptureWhiteBalanceModeContinuousAutoWhiteBalance;
+            break;
+    }
+
+    NSError *error;
+    if ([self.videoCaptureDevice lockForConfiguration:&error]) {
+        if ([self.videoCaptureDevice isWhiteBalanceModeSupported:whiteBlanceMode]) {
+            self.videoCaptureDevice.whiteBalanceMode = whiteBlanceMode;
+            _cameraWhiteBlanceMode = cameraWhiteBlanceMode;
+        }
+        [self.videoCaptureDevice unlockForConfiguration];
+    }
+
+    if (error) {
+        [self handleErrorWithCode:error.code userInfo:error.userInfo];
+    }
+}
+
+- (CGFloat)minExposureBias
+{
+    CGFloat minExposureBias;
```

Comment:
Good idea! I like your idea but I decide to put NAN be the default value. If this getter can't work correctly, the developer will get NAN value. The main reason that I don't want put NSInteger to CGFloat.